src/api/product-api/product-api-types.ts

```diff
@@ -0,0 +1,167 @@
+export type Product = {
+  createdAt: string;
+  createdBy: CreatedAndModified;
+  id: string;
+  key: string;
+  lastMessageSequenceNumber: number;
+  lastModifiedAt: string;
+  lastModifiedBy: CreatedAndModified;
+  lastVariantId: number;
+  masterData: MasterData;
+  productType: ProductInfo;
+  taxCategory: ProductInfo;
+  version: number;
+  versionModifiedAt: string;
+};
+
+export type AllProducts = {
+  count: number;
+  limit: number;
+  offset: number;
+  results: Product[];
+  total: number;
+};
+
+export type ProductCards = ProductCard[];
+
+export type ProductCard = {
+  description: string;
+  image: string;
+  name: string;
+  price: number;
+  discount?: number;
+  id: string;
+};
+
+export interface MyCustomEvent {
+  detail: string;
+  data: EventData[];
+  currentPage: number;
+  totalPages: number;
+  query: FilterProductsQuery;
+}
+
+type EventData = {
+  categories: Ids[];
+  categoryOrderHints: object;
+  createdAt: string;
+  description: Language;
+  hasStagedChanges: boolean;
+  id: string;
+  key: string;
+  lastModifiedAt: string;
+  masterVariant: MasterVariant;
+  metaDescription: {
```

Comment:
maybe use 1 option type for Localization  - type Locale = Record<LocaleKeys, string> ?